### ISSUE TITLE
`primary_key` field seems to be ignored. Leads to inconsistencies in errors K1-2

### ISSUE THREAD

**Issue by simonMeero** [looker-open-source/look-at-me-sideways#67]
Hello, I came across some unexpected behavior while running LAMS on my code.
I traced it back to what I _think_ are some issues in the code, but I'm not familiar with js (or with LookML for that matter), so I'm not 100% sure.

See the the `test.lkml` below; 
```
view: my_view {
  sql_table_name: my_db.table ;;
  dimension: something {
    primary_key: no
    hidden: yes
    type: number
    sql: ${TABLE}.id ;;
  }
}
```

Following the definitions of K1 and K2 as per `docs/rules.html`

- :heavy_check_mark:Running LAMS on this raises a `K1... No Primary Key Dimensions found in my_view`, which is the expected result.
- :heavy_check_mark:  Setting  `primary_key: yes` **and** renaming the dimension to  `pk1_something` raises no error, which is the expected result.
- :x:  **Only** setting  `primary_key: yes` still leads to a K1 error, while I would expect it to be a K2.
- :x:  **Only** renaming the dimension to  `pk1_something` raises no error, while I would expect a K1

Looking at the code, it looks like the `primary_key` field is never actually looked at. 
In `rules/k1-2-3-4.js`, primary keys are implicitly defined as fields matching the `pkNamingConvention` filter, even when `$dim.primary_key` is false. 

(The regex used in `pkNamingConvention` is itself not consistent with the documentation, see issue #63.)

(_If you are open to external PRs, I may have some time for this one_)

**Comment by simonMeero** [comment]
Hello, thank you for your answer.
I understand that a `primary_key` rule change would be a larger change more suited for a new version. Unfortunately, I'll only be involved in my LookML project for a short time, so I won't have the bandwidth (or skills) to be part of it.

However, I might have time to address issue 63, with tests, before the end of the week, I'll have to check that with the team. I also need to check whether our own codebase contains many multi-dimensional primary keys. If not, we may want to use the `pk_something` format indeed.

**Comment by simonMeero** [comment]
I opened the PR above, I just need to make sure it's okay for me to accept the CLA.

**Comment by fabio-looker** [comment]
@simonMeero Hey, I never saw your question in your previous comment (maybe due to the edit & me having read it before the edit? not sure), but in case it still helps...

> Do you mean that an table with a multi-column pk must have those column concatenated in a single pk dimension in the associated view ? I don't have much experience with looker so maybe I'm missing something.

Yes... not for the LAMS linter rules, but for the purposes of Looker's own validation. Looker uses `primary_key` information to correctly generate distinct sums for "symmetric aggregates" (i.e. distinct sum when trying to sum from  table that has been many-to-one joined, like a dimension table). However, Looker only allows `primary_key` to be applied to one field, so if your primary key has multiple columns, you have to generate one field that concatenates the columns and declare that as the `primary_key`. This declaration is not as useful for our style guide which benefits from separate fields, so you end up doing it both ways in the case of a composite primary key.

**Comment by fabio-looker** [comment]
I think this has been fixed since 2.1.3, so closing out the issue. Let me know if I missed anything!

Also, v3 made all rules opt-in, so if there are any new rules you would like to see added, the friction around adding them has been significantly reduced!
